text/0000-isHtmlSafe.md

```diff
@@ -0,0 +1,79 @@
+- Start Date: 2016-04-18
+- RFC PR: (leave this empty)
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Introduce `Ember.String.isHtmlSafe()` to provide a reliable way to determine if an object is an "html safe string", IE was it created with `Ember.String.htmlSafe()`.
+
+
+# Motivation
+
+Prior to the deprecation of `new Ember.Handlebars.SafeString()` you could easily test this with the following code:
+`value instanceof Ember.Handlebars.SafeString`. However, due to the technical details of the deprecation this test is
+no longer possible. Beyond that, the aforementioned test is a bit leaky. It requires the developer to understand
+`htmlSafe` returns a `Ember.Handlerbars.SafeString` instance and thus limits Ember's ability to change
+`htmlSafe` without further breaking it's API.
+
+Based on our app at Batterii and some research on Github, I see two valid use cases for introducing this API.
+
+First, and most commonly, is to make it possible to test addon helpers that are expected to return a safe string. I believe this test on ember-i18n says it all: ["returns HTML-safe string"](https://github.com/jamesarosen/ember-i18n/blob/master/tests/unit/utils/i18n/default-compiler-test.js#L56-L59).
+
+The second use case is to do type checking. In our app, we have an `isString` utility that is effectively:
+
+```javascript
+import Ember from 'ember';
+
+export default function(value) {
+  return typeof value === 'string' || value instanceof Ember.Handlebars.SafeString;
+}
+```
+
+Newer versions of ember-i18n, doing `this.get('i18n').t('someTranslatedValue')` will return a safe string. Thus our `isString` utility has to consider that.
+
+
+# Detailed design
+
+`htmlSafe` will be added to the `Ember.String` module. The implementation will basically be:
+
+```javascript
+function isHtmlSafe(str) {
+  return str instanceof SafeString;
```

Comment:
This should do what the rest of the internals do, and check `str && typeof str.toHTML === 'function'`.